### PR TITLE
Payment Verification of authorization transaction on subscription.

### PR DESCRIPTION
**Pull request by sanzzzay**
This fix will verify the signature of the payment done after the authorization transaction done on subscription.
https://razorpay.com/docs/api/subscriptions/#checkout-integration
https://razorpay.com/docs/api/subscriptions/#payment-verification

### CODE REVIEW

**Comment by sanzzzay** [comment]
Hey @razorpay-sanjib can you check this PR. If it looks good, please approve.

File: razorpay/utility/utility.py [hunk 1]
```diff
@@ -20,7 +20,21 @@ def verify_payment_signature(self, parameters):
...
+        """
+        To consider the payment as successful and subscription as authorized 
+        after the signature has been successfully verified
+        """
+        subscription_id = str(parameters['razorpay_subscription_id'])
+        payment_id = str(parameters['razorpay_payment_id'])
+        razorpay_signature = str(parameters['razorpay_signature'])
 
+        msg = "{}|{}".format(payment_id, subscription_id)
+        secret = str(self.client.auth[1])
```
Comment (razorpay-sanjib) [inline razorpay/utility/utility.py:34]: @sanzzzay The above questions applies here as well, this time the index `1` in `self.client.auth`

**Review by razorpay-sanjib — COMMENTED** [review]
@sanzzzay Please have a look at the comments.

File: razorpay/utility/utility.py [hunk 2]
```diff
@@ -20,7 +20,21 @@ def verify_payment_signature(self, parameters):
...
         return self.verify_signature(msg, razorpay_signature, secret)
+    
+    def verify_subscription_payment_signature(self, parameters):
+        """
+        To consider the payment as successful and subscription as authorized 
+        after the signature has been successfully verified
+        """
+        subscription_id = str(parameters['razorpay_subscription_id'])
+        payment_id = str(parameters['razorpay_payment_id'])
+        razorpay_signature = str(parameters['razorpay_signature'])
```
[the remark this answers is not in the document]
Comment (sanzzzay) [reply razorpay/utility/utility.py:31]: @razorpay-sanjib here I just kept the code just like existing method `verify_payment_signature`
I have kept the same utility method in my code where I am handling `SignatureVerificationError` exception raised by `verify_signature` .

File: razorpay/utility/utility.py (hunk 1 above)
Comment (sanzzzay) [reply razorpay/utility/utility.py:34]: I have written a utility method in my code `get_client` where I am passing API_KEY and API_SECRET and handling the exception so won't encounter this issue.

**Review by razorpay-sanjib — APPROVED** [review]
Looks good to me

**Comment by sanzzzay** [comment]
@captn3m0 @neera11 or anyone with write access to this repo. Please merge this PR.
Thanks in advance.

**Comment by sanzzzay** [comment]
@razorpay-sanjib can you help me with the merging process of this PR?

**Comment by razorpay-sanjib** [comment]
> @razorpay-sanjib can you help me with the merging process of this PR?

@sanzzzay done

**Comment by sanzzzay** [comment]
> > @razorpay-sanjib can you help me with the merging process of this PR?
> 
> @sanzzzay done

Thanks for the quick response. Much appreciated.

**Pull request merged**